politeiawww/backend.go

```diff
@@ -1686,61 +1671,87 @@ func (b *backend) ProcessProposalDetails(propDetails www.ProposalsDetails, user
 		changes:  p.changes,
 		comments: p.comments,
 	}, b.userPubkeys)
-	reply.Proposal.Username = b.getUsernameById(reply.Proposal.UserId)
-
 	return &reply, nil
 }
 
-// ProcessComment processes a submitted comment.  It ensures user has paid
-// the paywall, and the proposal and the parent exists.  A parent ID of 0
-// indicates that it is a comment on the proposal whereas non-zero
-// indicates that it is a reply to a comment.
+// ProcessComment processes a submitted comment.  It ensures the proposal and
+// the parent exists.  A parent ID of 0 indicates that it is a comment on the
+// proposal whereas non-zero indicates that it is a reply to a comment.
 func (b *backend) ProcessComment(c www.NewComment, user *database.User) (*www.NewCommentReply, error) {
 	log.Debugf("ProcessComment: %v %v", c.Token, user.ID)
 
+	// Pay up sucker!
 	if !b.VerifyUserPaid(user) {
 		return nil, www.UserError{
 			ErrorCode: www.ErrorStatusUserNotPaid,
 		}
 	}
 
+	// Verify authenticity.
 	err := checkPublicKeyAndSignature(user, c.PublicKey, c.Signature,
 		c.Token, c.ParentID, c.Comment)
 	if err != nil {
 		return nil, err
 	}
 
-	b.Lock()
-	defer b.Unlock()
-	m, ok := b.inventory[c.Token]
-	if !ok {
-		return nil, www.UserError{
-			ErrorCode: www.ErrorStatusProposalNotFound,
-		}
+	// Validate comment
+	if err := validateComment(c); err != nil {
+		return nil, err
 	}
 
-	// See if we are commenting on a comment, yo dawg.
-	if c.ParentID == "" {
-		// "" means top level comment; we need it to be "0" for the
-		// underlying code to understand that.
-		c.ParentID = "0"
+	// Setup plugin command
+	challenge, err := util.Random(pd.ChallengeSize)
+	if err != nil {
+		return nil, err
 	}
-	pid, err := strconv.ParseUint(c.ParentID, 10, 64)
+
+	ndc := convertWWWNewCommentToDecredNewComment(c)
+	payload, err := decredplugin.EncodeNewComment(ndc)
 	if err != nil {
-		return nil, www.UserError{
-			ErrorCode: www.ErrorStatusCommentNotFound,
-		}
+		return nil, err
 	}
-	if pid != 0 {
-		_, ok = m.comments[pid]
-		if !ok {
-			return nil, www.UserError{
-				ErrorCode: www.ErrorStatusCommentNotFound,
-			}
-		}
+
+	pc := pd.PluginCommand{
+		Challenge: hex.EncodeToString(challenge),
+		ID:        decredplugin.ID,
+		Command:   decredplugin.CmdNewComment,
+		CommandID: decredplugin.CmdNewComment,
```

Comment:
Is this a typo that the Command and CommandID are the same?

Reply:
No. You keep asking this question too :-)

The reason it exists is in case the client needs to be able to have some sort of unique id to identify a response. We should be able to remove it if you feel strongly about it. Old protocols always dealt with disconnects and I tend to add this so that it does not have to be added later. This being JSON over HTTPS I am ok killing it.